sdk/compatability/src/lib.rs

```diff
@@ -0,0 +1,592 @@
+    env.coffer()
+        .fund(currency, account.authentication_key(), start_amount)?;
+
+    let mut account_2 = env.random_account();
+    env.coffer()
+        .fund(currency, account_2.authentication_key(), start_amount)?;
+
+    // Sending a txn that's over the dual_attestation_limit without a signature from the reciever
+    let txn = account.sign_with_transaction_builder(env.transaction_factory().peer_to_peer(
+        currency,
+        account_2.address(),
+        transfer_amount,
+    ));
+
+    client.submit(&txn)?;
+    assert!(matches!(
+        client
+            .wait_for_signed_transaction(&txn, None, None)
+            .unwrap_err(),
+        diem_sdk::client::WaitForTransactionError::TransactionExecutionFailed(_),
+    ));
+
+    // Add a dual_attestation creds for account_2
+    let dual_attestation_secret_key = ed25519_dalek::SecretKey::generate(&mut rand::rngs::OsRng);
+    let dual_attestation_extended_secret_key =
+        ed25519_dalek::ExpandedSecretKey::from(&dual_attestation_secret_key);
+    let dual_attestation_public_key = ed25519_dalek::PublicKey::from(&dual_attestation_secret_key);
+
+    let txn = account_2.sign_with_transaction_builder(
+        env.transaction_factory().rotate_dual_attestation_info(
+            "https://example.com".as_bytes().to_vec(),
+            dual_attestation_public_key.as_bytes().to_vec(),
+        ),
+    );
+
+    client.submit(&txn)?;
+    client.wait_for_signed_transaction(&txn, None, None)?;
+
+    let metadata = "metadata";
+    let sig = {
+        let message =
+            DualAttestationMessage::new(metadata.as_bytes(), account.address(), transfer_amount);
+        dual_attestation_extended_secret_key
+            .sign(message.message(), &dual_attestation_public_key)
+            .to_bytes()
+            .to_vec()
+    };
+    // Send a p2p txn with the signed metadata
+    let txn = account.sign_with_transaction_builder(
+        env.transaction_factory().peer_to_peer_with_metadata(
+            currency,
+            account_2.address(),
+            transfer_amount,
+            metadata.as_bytes().to_vec(),
+            sig,
+        ),
+    );
+
+    client.submit(&txn)?;
+    client.wait_for_signed_transaction(&txn, None, None)?;
+
+    let account_view_1 = client.get_account(account.address())?.into_inner().unwrap();
+    let balance = account_view_1
+        .balances
+        .iter()
+        .find(|b| b.currency == currency)
+        .unwrap();
+    assert_eq!(balance.amount, start_amount - transfer_amount);
+
+    let account_view_2 = client
+        .get_account(account_2.address())?
+        .into_inner()
+        .unwrap();
+    let balance = account_view_2
+        .balances
+        .iter()
+        .find(|b| b.currency == currency)
+        .unwrap();
+    assert_eq!(balance.amount, start_amount + transfer_amount);
+
+    Ok(())
+}
+
+#[test]
+#[ignore]
+fn no_dual_attestation_needed_with_child_account() -> Result<()> {
```

Comment:
LoL. First I thought this is a bug.
Looking into details, it looks like transfer between parent vasp and it's child vasp has no dual attestation, which is fine.
may update the test name to include something like: between parent vasp and it's child vasps.
It should do dual attestation when:
1. p2p between a child vasp and a different parent vasp
2. p2p between a child vasp and another child vasp that has different parent vasp

Reply:
I'll update the name of the test